Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/onboarding/OnboardingAnalyticsViewModel.kt

```diff
@@ -32,15 +33,24 @@ class OnboardingAnalyticsViewModel @AssistedInject constructor(
     }.asLiveData(dispatcherProvider.IO)
 
     fun onNextButtonClick() {
+        settings.analyticsEnabled.update {
+            true
+        }
+
         completedOnboardingEvent.postValue(Unit)
-        // TODO: ♫ ♪ Hello Kolya, my old friend ♫ ♪
-        // TODO: ♫ ♪ Place boolean = TRUE at the end ♫ ♪
     }
 
     fun onDisableClick() {
-        settings.userInfoAgeGroup.update { PpaData.PPAAgeGroup.AGE_GROUP_UNSPECIFIED }
-        settings.userInfoFederalState.update { PpaData.PPAFederalState.FEDERAL_STATE_UNSPECIFIED }
-        settings.userInfoDistrict.update { 0 }
+        settings.analyticsEnabled.update {
+            false
+        }
+
+        launch {
```

Comment:
If the user smashes that like button, err disable button.... multiple presses can launch this multiple times, right? So donor modules would have to be thread safe on `deleteData`.

Reply:
hmm true, i am also not 100% happy with the positioning here, as we would need to duplicate the code in the AnalyticsSettings view model later on, maybe also some kind of mutex lockout and we move this to a more generic place where it listens to the analyticsEnabled flow preference?

Reply:
Note that FlowPreference is only safe when accessed from the same thread. It's not safe for concurrent access from multiple threads.

I think it makes sense if all deletions, just like starting the analytics submission goes the through `Analytics` main class, and we add a single `Mutex' that guards the deletion and the submission call (as those should not overlap either).

What do you think?

Reply:
sounds good, i've wrapped changes to the property in calls on `Analytics` which should allow us to execute delete on every settings change